### ISSUE TITLE
ghidra failed to disassembly and decompile

### ISSUE THREAD

**Issue by RocketMaDev** [NationalSecurityAgency/ghidra#5825]
**Describe the bug**

I just analyzed a elf file, which will be attached with ghidra, but ghidra could not disassembly properly and made it hard to read. Even external funtion calls are processed as local functions with improper name. I'm a ctf player in pwn direction, so the program DOES have bugs. The screenshot of BinaryNinja showed the proper decompiling result.

**To Reproduce**

Just auto analyze the file.

**Expected behavior**

Like the screenshot of BinaryNinja.

**Screenshots**

![unreadable result](https://github.com/NationalSecurityAgency/ghidra/assets/48864431/b80c5fe6-1d9e-44eb-bacb-97621025d854)
![expected result](https://github.com/NationalSecurityAgency/ghidra/assets/48864431/15d223ec-a2cd-4ae0-b231-7c778381d412)

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

[format_level0.zip](https://github.com/NationalSecurityAgency/ghidra/files/12715537/format_level0.zip)

**Environment (please complete the following information):**
 - OS: Arch Linux 6.5.2
 - Java Version: 17.0.8 Graal VM
 - Ghidra Version: 10.3.3
 - Ghidra Origin: official GitHub distro

**Additional context**

The binary belongs to a ctf match organization which is holding a match till 14th Oct. Don't spread it.

**Comment by ghidra1** [comment]
It appears we do not currently handle the PLT thunking scheme employed by this binary.  This will take some effort on our part to add support for.  I will submit as an internal ticket to address although I can't comit to a timeframe.  Thank you for reporting this and providing the sample.

**Comment by ghidra1** [comment]
You can try the following steps to get the thunks to resolve in the `.plt.got` and `.plt.sec` sections which involves setting the EBX value equal to the address of `.got` section over the sections.  For the sample provided this is 0x13fac.
```
1) Select the entire section (e.g., `.plt.got`)
2) Invoke the Set Register Values... popup menu action and assign EBX value as 0x13fac
3) With selection still in place, invoke Analysis->Auto Analyze action (thunks should resolve)
4) Repeat for other EBX based PLT thunks (e.g., .plt.sec)
```

**Comment by ghidra1** [comment]
I had also altered the `init` function signature which may have been improperly set.  I assumed this since it did not match the signature which had been produced by DWARF and appears to have been applied from one of the datatype archives.

**Comment by RocketMaDev** [comment]
I've checked your solution and it works perfectly. But what confused me is that why other binaries are not effected, whose `.plt.got` and `.plt.sec` sections are not assigned a ebx value, either.

BTW, could you inform me of what is the font id of the font when I open `set pre comments` dialog in theme?

**Comment by ghidra1** [comment]
In most cases we rely on a constant propogation analyzer to compute the destination address which in this case is indirect and requires a valud for EBX.  There are numerous GOT/PLT mechanisms for each processor and most do not utilize a global register value such as this.  The ultimate fix for this case when handled by the Ghidra ELF Loader will likely require setting the EBX value as I have outlined.  The rest is handled by analysis.

Regarding the UI font for the Comment Dialog, it appears that there is currently a hard-coded Monospaced font forced for JTextArea.  Thanks for pointing out this concern.  A fix for this is in the works.  Since this dialog does not use any custom IDs, it relies on the default for `laf.font.JTextArea.font` within the theme editor.  At present, the hard-coded font setting (not reflected by the theme editor) is overriding the Theme setting.

**Comment by RocketMaDev** [comment]
One more suggestion: please enlarge the icons, if you have time. Recently, more and more developers use hi-res screens, old icons are a bit difficult to click on since they are too small; if just relying on the system to enlarge icons, they looks vague. So providing a icon pack and adding a button to load it in Theme Config dialog seems to be a good choice.

Anyway, thank you, @ghidra1 , for solving my problem. The improvement in suggestion is optional, whether implementing it or not depends on you (or your team).

**Comment by mike-hunhoff** [comment]
> You can try the following steps to get the thunks to resolve in the `.plt.got` and `.plt.sec` sections which involves setting the EBX value equal to the address of `.got` section over the sections. For the sample provided this is 0x13fac.
> 
> ```
> 1) Select the entire section (e.g., `.plt.got`)
> 2) Invoke the Set Register Values... popup menu action and assign EBX value as 0x13fac
> 3) With selection still in place, invoke Analysis->Auto Analyze action (thunks should resolve)
> 4) Repeat for other EBX based PLT thunks (e.g., .plt.sec)
> ```
> 
> After doing this I achieved the following result for `main` method decompilation: ![image](https://private-user-images.githubusercontent.com/48333017/271000660-cc342439-5b63-4f35-96ae-56f606becd64.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2MDkzMjAsIm5iZiI6MTcxNzYwOTAyMCwicGF0aCI6Ii80ODMzMzAxNy8yNzEwMDA2NjAtY2MzNDI0MzktNWI2My00ZjM1LTk2YWUtNTZmNjA2YmVjZDY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDE3MzcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxZjNjYTljMGM0MTBkZTllMTAxNzU5MjNmOWQ3Zjk4NTczOWM2ZDg2OTAzNTUyM2VjMGRlOWRkOGUxN2YwM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qBiNylKyrtkQcLrTIywzX5ZxPSAAerMElbOUyycv2So)

@ghidra1 I recently encountered this issue in Ghidra 11.0.3 - is there a planned fix?

**Comment by mike-hunhoff** [comment]
> > You can try the following steps to get the thunks to resolve in the `.plt.got` and `.plt.sec` sections which involves setting the EBX value equal to the address of `.got` section over the sections. For the sample provided this is 0x13fac.
> > ```
> > 1) Select the entire section (e.g., `.plt.got`)
> > 2) Invoke the Set Register Values... popup menu action and assign EBX value as 0x13fac
> > 3) With selection still in place, invoke Analysis->Auto Analyze action (thunks should resolve)
> > 4) Repeat for other EBX based PLT thunks (e.g., .plt.sec)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > After doing this I achieved the following result for `main` method decompilation: ![image](https://private-user-images.githubusercontent.com/48333017/271000660-cc342439-5b63-4f35-96ae-56f606becd64.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2MDkzMjAsIm5iZiI6MTcxNzYwOTAyMCwicGF0aCI6Ii80ODMzMzAxNy8yNzEwMDA2NjAtY2MzNDI0MzktNWI2My00ZjM1LTk2YWUtNTZmNjA2YmVjZDY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDE3MzcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxZjNjYTljMGM0MTBkZTllMTAxNzU5MjNmOWQ3Zjk4NTczOWM2ZDg2OTAzNTUyM2VjMGRlOWRkOGUxN2YwM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qBiNylKyrtkQcLrTIywzX5ZxPSAAerMElbOUyycv2So)
> 

@ghidra1 I recently encountered this issue in Ghidra 11.0.3 - is there a planned fix? The manual steps that you outlined worked but these are not great for automation.

**Comment by ghidra1** [comment]
I have a fix in for Ghidra 11.1.1 patch release

**Comment by ryanmkurtz** [comment]
Fixed by b1a67cce480e2223b0d6e2472f82d6c8a9bb6e48
